0.3/freefare/ultralight.go

```diff
@@ -101,3 +101,50 @@ func (t UltralightTag) Authenticate(key DESFireKey) error {
 	}
 	return t.TranslateError(err)
 }
+
+// Set the provided authentication key. Note that this only works with
+// MifareUltralightC tags. It _should_ work only after authentication,
+// but for some reason the opposite is true: it only works without it.
+func (t UltralightTag) SetKey(key DESFireKey) error {
+	r, err := C.mifare_ultralightc_set_key(t.ctag, key.key)
+	if r == 0 {
+		return nil
+	}
+
+	return t.TranslateError(err)
+}
+
+// Returns key derivation object
+func (t UltralightTag) NewAn10922(key DESFireKey, keyType MifareKeyType) MifareKeyDeriver {
+	return MifareKeyDeriver{t.tag, C.mifare_key_deriver_new_an10922(key.key, C.MifareKeyType(keyType), 0)}
```

Comment:
This is a good example of what I'm talking about: when I was working on this, the signature was `mifare_key_deriver_new_an10922(MifareDESFireKey master_key, MifareKeyType output_key_type)`. Now it's `mifare_key_deriver_new_an10922(MifareDESFireKey master_key, MifareKeyType output_key_type, int flags)`, with `flags` being either `#define AN10922_FLAG_DEFAULT 0` or '#define AN10922_FLAG_EMULATE_ISSUE_91 (1<<1)`. Why do they feel this is the kind of API change that should be on master branch -- I don't know, but that's how it is.

Reply:
Yeah, I've previously [complained about this sort of thing](https://github.com/nfc-tools/libfreefare/issues/46), too.  They haven't cut a 1.0 release for four years and somehow think this gives them a license to do as they wish with respect to compatibility.